napari/plugins/_npe2.py

```diff
@@ -488,29 +504,183 @@ def _add_sample(
     return submenu, sample_actions
 
 
+def _get_widget_param(widget_callable, widget_name) -> str:
```

Comment:
would like a more specific name here like `_satisfy_widget_viewer_param` or something? It doesn't look like it deals with any other parameters

Reply:
Would also be nice to get more coverage on this function as it's pretty critical and I worry about it silently breaking plugins. I think we can get coverage relatively easily by expanding on the tests in `test_plugin_widgets`?

Reply:
It's not satisfying it (that happens in the callback) we are just getting the parameter name, but yes I think we decided that plugins will only get viewer provided so will update name to `_get_widget_viewer_param`.

I've added unit tests for `_get_widget_viewer_param`, let me know what you think!